File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/enums_runme.lua
require("import")	-- the import fn
import("enums")	-- import lib

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

-- check values
assert(enums.CSP_ITERATION_FWD==0)
assert(enums.CSP_ITERATION_BWD==11)
assert(enums.ABCDE==0)
assert(enums.FGHJI==1)
assert(enums.boo==0)
assert(enums.hoo==5)
assert(enums.globalinstance1==0)
assert(enums.globalinstance2==1)
assert(enums.globalinstance3==30)
assert(enums.AnonEnum1==0)
assert(enums.AnonEnum2==100)

-- In C enums from struct are exported into global namespace (without prefixing with struct name)
-- In C++ they are prefixed (as compatibility thing).
-- We are emulating xor :)
assert(enums.BAR1 ~= enums.Foo_BAR1) -- It is either C style, or C++ style, but not both
assert((enums.BAR1 ~= nil ) or (enums.Foo_BAR1 ~= nil))

assert(enums.Phoo ~= enums.iFoo_Phoo)
assert((enums.Phoo == 50) or (enums.iFoo_Phoo == 50))
-- no point in checking fns, C will allow any value

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/enums_runme.rb
#!/usr/bin/env ruby
#
# Runtime tests for enums.i
#

require 'swig_assert'
require 'enums'

swig_assert_each_line( <<EOF )
Enums::CSP_ITERATION_FWD == 0
Enums::CSP_ITERATION_BWD == 11
Enums::ABCDE == 0
Enums::FGHJI == 1
Enums.bar1(1)
Enums.bar2(1)
Enums.bar3(1)
Enums::Boo == 0
Enums::Hoo == 5
Enums::Globalinstance1 == 0
Enums::Globalinstance2 == 1
Enums::Globalinstance3 == 30
Enums::AnonEnum1 == 0
Enums::AnonEnum2 == 100
Enums::BAR1 == 0
Enums::BAR2 == 1
EOF

#
# @bug: 
#
# swig_assert_each_line( <<EOF )
# Enums::IFoo::Phoo == 50
# Enums::IFoo::Char == 'a'[0]
# EOF
